common/src/Microsoft.Azure.IIoT.Agent.Framework/src/Jobs/Hub/JobOrchestratorEndpointSync.cs

```diff
@@ -52,40 +58,51 @@ public class JobOrchestratorEndpointSync : IHostProcess, IDisposable {
         public Task StopAsync() {
             if (_cts != null) {
                 _cts.Cancel();
-                _updateTimer.Change(0, Timeout.Infinite);
             }
+
+            lock(_timerLock) {
+                // Disabling future invocation of timer callback.
+                _updateTimer.Change(Timeout.Infinite, Timeout.Infinite);
+            }
+
             return Task.CompletedTask;
         }
 
         /// <summary>
         /// Timer operation
         /// </summary>
         /// <param name="sender"></param>
-        private async void OnUpdateTimerFiredAsync(object sender) {
-            try {
-                _cts.Token.ThrowIfCancellationRequested();
-                _logger.Information("Updating orchestrator urls...");
-                await UpdateOrchestratorUrlAsync(_cts.Token);
-                _logger.Information("Orchestrator Url update finished.");
-            }
-            catch (OperationCanceledException) {
-                // Cancel was called - dispose cancellation token
-                _cts.Dispose();
-                _cts = null;
-                return;
-            }
-            catch (Exception ex) {
-                _logger.Error(ex, "Failed to update orchestrator urls.");
+        private void OnUpdateTimerFiredAsync(object sender) {
```

Comment:
now its not async...

Reply:
Yes, `lock() { ... }` doesn't allow me to have `async` operations with `await`, so I had to change all `await`s to `.Wait()`.